### PR TITLE
new style minimum function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `minimum` function.

### CODE REVIEW

**Comment by takagi** [comment]
Travis fails on macOS Python 3.4 case at least because of the problem fixed in #3599. Would you rebase the master?

**Comment by kmaehashi** [comment]
Rebased.

File: chainer/functions/math/minimum.py [hunk 1]
```diff
@@ -46,6 +56,13 @@ def backward_gpu(self, inputs, grads):
             'minimum_bwd2')(x1, x2, gy)
         return gx1, gx2
 
+    def backward(self, indexes, grad_outputs):
+        x1, x2 = self.get_retained_inputs()
+        ggx = grad_outputs[0] * 0 + grad_outputs[1] * 0
```
Comment (takagi) [inline chainer/functions/math/minimum.py]: Name `gx` would be better?

File: chainer/functions/math/minimum.py [hunk 2]
```diff
@@ -46,6 +56,13 @@ def backward_gpu(self, inputs, grads):
...
+        cond = utils.force_array(x1.data <= x2.data)
+        gggy = chainer.functions.where(cond, grad_outputs[0], grad_outputs[1])
```
Comment (takagi) [inline chainer/functions/math/minimum.py]: Name `ggy` would be better?

File: chainer/functions/math/minimum.py (hunk 1 above)
Comment (takagi) [reply chainer/functions/math/minimum.py]: Memory spaces are allocated for each expression result here?

**Comment by takagi** [comment]
Would you rebase the master?

File: chainer/functions/math/minimum.py [hunk 3]
```diff
@@ -18,24 +19,33 @@ def check_type_forward(self, in_types):
         )
 
     def forward_cpu(self, inputs):
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply chainer/functions/math/minimum.py]: Fixed.

**Comment by kmaehashi** [comment]
Thanks for review, rebased and fixed accordingly.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**